dot/rpc/websocket_test.go

```diff
@@ -48,15 +48,15 @@ var testCalls = []struct {
 	{
 		call:     []byte(`{"jsonrpc":"2.0","method":"chain_subscribeFinalizedHeads","params":[],"id":5}`),
 		expected: []byte(`{"jsonrpc":"2.0","result":3,"id":5}` + "\n")},
+	//{
```

Comment:
@timwu20 This is the code that is breaking the unit tests. Feels weird that this is testing for an error (my guess is that it was made when txn validation didn't work. Not sure what caused it to break now, but since I fix txn stuff in other PR can I delete this? I tried providing extrinsic as a parameter that passed for other tests but it wasn't working for some reason

Reply:
as mentioned in our chat, the test is timing out on this case now.  You'll need to investigate why there are no messages being sent back over the websocket after sending a message with `author_submitAndWatchExtrinsic` for the method.

Reply:
@timwu20 Done. Since we changed the unmarshalling of a txnValidityErr to return either an invalid or unknown txn (rather than the outer vdt as I had initially) it was causing a failed typecast to the outer vdt and thus not sending a response, causing it to hang. Should be all good now

Reply:
This actually made me realize I did the same thing in core/message, so adding a fix for that now as well